storage/storage.go

```diff
@@ -0,0 +1,96 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/golang-migrate/migrate"
+	migratepg "github.com/golang-migrate/migrate/database/postgres"
+	_ "github.com/golang-migrate/migrate/source/file"
+	_ "github.com/lib/pq"
+	"log"
+	"path"
+	"runtime"
+)
+
+var (
+	_, file, _, _ = runtime.Caller(0)
+	basepath      = path.Dir(file)
+)
+
+// Config holds all storage configuration settings
+type Config struct {
+	Type string
+	URI  string
+}
+
+type Storage interface {
+	Get() *sql.DB
+	Transaction(context context.Context, operation func(context context.Context, tx *sql.Tx) error) error
+	Close() error
+}
+
+type postgresStorage struct {
+	db *sql.DB
+}
+
+func (ps *postgresStorage) Get() *sql.DB {
+	return ps.db
+}
+
+func (ps *postgresStorage) Transaction(context context.Context, operation func(context context.Context, tx *sql.Tx) error) error {
+	tx, err := ps.db.BeginTx(context, nil)
+	if err != nil {
+		return err
+	}
+
+	if opErr := operation(context, tx); opErr != nil {
+		if err := tx.Rollback(); err != nil {
+			return err
+		}
+		return opErr
+	}
+	return tx.Commit()
+}
+
+func (ps *postgresStorage) Close() error {
+	return ps.db.Close()
+}
+
+// New creates a Storage object and updates the database with the latest migrations
+func New() (Storage, error) {
+	db, err := sql.Open("postgres", "postgres://postgres:postgres@localhost:1234/postgres?sslmode=disable")
+	if err != nil {
+		return nil, fmt.Errorf("unable to open db connection: %s", err)
+	}
+
+	err = updateSchema(db, "postgres")
+	if err != nil {
+		return nil, fmt.Errorf("unable to update db schema: %s", err)
+	}
+
+	log.Println("Database is up-to-date")
```

Comment:
Use info level of logging

Reply:
Not applicable for log build in package.